estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,49 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, exceptions
+from odoo.tools import relativedelta
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Real Estate Property Offer"
+
+    price = fields.Float()
+    status = fields.Selection([
+        ('accepted', 'Accepted'),
+        ('refused', 'Refused'),
+    ], copy=False)
+
+    partner_id = fields.Many2one('res.partner', required=True)
+    property_id = fields.Many2one('estate.property', required=True)
+
+    validity = fields.Integer(string="Offer Validity (days)", default=7)
+    date_deadline = fields.Date(string="Deadline", compute="_compute_deadline_date", inverse="_inverse_deadline_date")
+
+    @api.depends('validity')
+    def _compute_deadline_date(self):
+        for record in self:
+            if record.create_date:
+                record.date_deadline = fields.Date.to_date(record.create_date) + relativedelta(days=record.validity)
+            else:
+                record.date_deadline = fields.Date.today() + relativedelta(days=record.validity)
+
+    def _inverse_deadline_date(self):
+        for record in self:
+            if record.create_date:
+                record.validity = (record.date_deadline - fields.Date.to_date(record.create_date)).days
+            else:
+                record.validity = (record.date_deadline - fields.Date.today()).days
+
+    def action_accept(self):
+        for record in self:
+            if record.property_id.offer_ids.filtered(lambda r: r.status == 'accepted'):
+                raise exceptions.UserError("Another offer is already accepted on this property.")
+            record.status = 'accepted'
+            record.property_id.selling_price = record.price
+            record.property_id.buyer_id = record.partner_id
```

Comment:
Don't you want to also update the state of the `estate.property`record ? Also, it is debatable but I don't like that we change multiple values on a foreign model. IMO, it'd be better to have a method on `estate.property` that we could call from here and would manage the state of the record itself.